component/src/ibc/component/client.rs

```
@@ -135,22 +141,40 @@ pub(crate) trait Ics2ClientExt: StateWrite {
     // - client type
     // - consensus state
     // - processed time and height
-    async fn execute_create_client(&mut self, msg_create_client: &MsgCreateAnyClient) {
+    async fn execute_create_client(&mut self, msg_create_client: &MsgCreateClient) {
+        tracing::info!("deserializing client state");
+        // TODO(erwan): deferred client state deserialization means `execute_create_client` is faillible
+        // see ibc-rs ADR004: https://github.com/cosmos/ibc-rs/blob/main/docs/architecture/adr-004-light-client-crates-extraction.md#light-client-specific-code
+        let client_state = match msg_create_client.client_state.type_url.as_str() {
+            TENDERMINT_CLIENT_STATE_TYPE_URL => {
+                TendermintClientState::try_from(msg_create_client.client_state.clone())
+                    .expect("decoding the tendermint client state")
+            }
+            _ => unimplemented!("not a tendermint lightclient"),
+        };
```

similar to the above, can we return an error here?